databases/elasticsearch/Portfile

```diff
@@ -15,31 +16,35 @@ long_description    Elasticsearch is a highly scalable open-source full-text \
                     real time. It is generally used as the underlying \
                     engine/technology that powers applications that have complex \
                     search features and requirements. \
-                    Elasticsearch requires Java, Java version can be up to Java 8. \
-                    Java 9 is not supported and Elasticsearch won't work.
+                    Elasticsearch requires OpenJDK, which comes installed. \
+                    Other Java version 9 and above are not supported.
 
 platforms           darwin
 supported_archs     noarch
 license             Apache-2
-
 homepage            https://www.elastic.co/
-master_sites        https://www.elastic.co/downloads/elasticsearch/
 
-checksums           rmd160  a6d687bcbae51ca42c43263c7ccff9e36c6b09e9 \
-                    sha256  6cc60fc08e8d40b25357d0b4d464e15fba803aad075c00c9b42bc6d578fd7f7e \
-                    size    28497714
+master_sites        https://artifacts.elastic.co/downloads/${name}/
+worksrcdir          ${distname}
+distname            ${name}-${version}-darwin-x86_64
+
+checksums           rmd160  c50a7a2e66e53b5df70afde3da2f971d1ac73886 \
+                    sha256  283859570434cd64146c61e5263c2cb3ce537e0dec567c39e0bc41e507eefed1 \
+                    size    338459414
 
-use_zip             yes
 use_configure       no
 build {}
 
+depends_run         port:openjdk12
```

Comment:
please use the java PortGroup, rather than hardcode this dependency. See for instance

https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/java-1.0.tcl

In this case

```
PortGroup        java         1.0
# Required java version.
java.version          12
# JDK port to install if required java not found
java.fallback         openjdk12
```

is probably all you need.

Reply:
What exactly does this do? After the licensing and compatibility issues with Oracle’s JDK v12, I’ve personally done a complete purge and moved everything entirely to openjdk12. That’s apparently what Elastic does because they bundle an openjdk12 within this release and their script adds it as `JAVA_HOME` as the default.

Because MP already has an `openjdk12`, I included this port as a dependency and omitted the copy that comes bundled within Elastic’s release.

Please advise.

Reply:
It allows for the possibility that the user may have installed it themselves, outside macports, and in that case will use it. Macports port will then only be installed if it is not already there.

Reply:
Thank you. Elastic’s support matrix says that they support both equally, so your mod is perfect. I will push it.